Mathlib/Algebra/BigOperators/Fin.lean

```diff
@@ -262,6 +262,65 @@ theorem inv_partialProd_mul_eq_contractNth {G : Type*} [Group G] (g : Fin (n + 1
 
 end PartialProd
 
+section DivModSum
+
+/-- Given a tuple of natural numbers `n : Fin m → ℕ` and a natural number `k`, return the first
+index `i` such that `k` is smaller than `∑ j < i, n j`, and `none` otherwise.
+
+This is defined en-route to `Fin.divSum`, which is the dependent version of `Fin.divNat`.
+-/
+def divSum? {m : ℕ} (n : Fin m → ℕ) (k : ℕ) : Option (Fin m) :=
+  find (fun i => k < ∑ j, n (castLE i.isLt j))
```

Comment:
This looks wrong to me; `isLt` is a proof of `<`, but `castLE` is about `≤`.  Can you make the type of `i` and `j` explicit here to make it clearer what is going on?

Reply:
So what happens here is that `i.isLt : i.val < n` is definitionally equal to `i.val.succ <= n`. So the sum is over `j : Fin i.val.succ`.